### PR TITLE
Fix file paths for Zip CSV imports 

### PR DESCRIPTION
**Pull request by bkiahstroud**
Only extract zip files once, instead of attempting to extract them for every Entry created

### CODE REVIEW

**Comment by orangewolf** [comment]
@bkiahstroud can you look to see if this code is still relevant?

**Comment by alishaevn** [comment]
## TLDR; importing a zip file only creates the files once so we're able to round trip csv's now. 🎉 

in testing round tripping for BL I discovered that importing a csv that had been exported with a type of "work type" or "collection" failed with `Zip::DestinationFileExistsError`. (see image below) I spoke with @orangewolf and @sephirothkod about it and we noticed that this pr was still sitting in draft state.

I reached out to @bkiahstroud to see if he remembered the direction he was headed in when he started this pr 9 months ago.

> I think the problem was when/where we were unzipping the file, like my commit message mentions. I think it tries to unzip it every time it creates an entry, and as soon as it hits the second entry it throws the error saying the file already exists because it was unzipped when making the first entry
I think this is also why the counters for zip imports are always in the thousands even if there’s only 100 rows
I think the idea I had was to unzip the zip file once, when the importer first starts, then just go look at it when creating the entries 

I pulled his branch and merged in the current version of `main`. I tested a combination of changes he had done and changes that were currently in `main` until I found the right combo. I'm now able to round trip the "work type" and "collection" export types! 🎉 

image

File: app/models/bulkrax/importer.rb [hunk 1]
```diff
@@ -97,7 +97,7 @@ def schedulable?
     end
 
     def current_run
-      @current_run ||= self.importer_runs.create!(total_work_entries: self.limit || parser.total, total_collection_entries: parser.collections_total)
```
Comment (orangewolf) [inline app/models/bulkrax/importer.rb:100]: it seems like this will break total reporting for anything that is not a zip file.

File: spec/models/bulkrax/importer_spec.rb [hunk 2]
```diff
@@ -21,14 +21,20 @@ module Bulkrax
     end
 
     describe 'importer run' do
-      before do
-        allow_any_instance_of(Bulkrax::OaiDcParser).to receive(:collections_total).and_return(1)
-      end
-
-      it 'creates an ImporterRun with total_work_entries set to the value of limit' do
+      it 'finds or creates a new ImporterRun' do
```
Comment (orangewolf) [inline spec/models/bulkrax/importer_spec.rb:24]: I think this test is likely wrong now (see comment above)

**Review by orangewolf — CHANGES_REQUESTED** [review]

File: app/models/bulkrax/importer.rb (hunk 1 above)
Comment (alishaevn) [reply app/models/bulkrax/importer.rb:100]: good point. I made this a conditional instead and tested each of the following successfully:
- import a csv of a work with no files
- import of a csv of a work with remote files
- import a zip of a csv with a "files" folder (local file)
- export by work type
- export by collection
- export by importer
- round trip of work type export
- round trip of collection export
- round trip of an importer export

File: spec/models/bulkrax/importer_spec.rb (hunk 2 above)
Comment (alishaevn) [reply spec/models/bulkrax/importer_spec.rb]: fixed.

File: app/models/bulkrax/csv_entry.rb [hunk 3]
```diff
@@ -79,13 +79,17 @@ def build_export_metadata
...
+        self.parsed_metadata['collection'] = hyrax_record.member_of_collection_ids.join('; ')
+        #   self.parsed_metadata['collection'] = hyrax_record.member_of_collections.map { |c| c.send(work_identifier)&.first }.compact.uniq.join(';')
       else
         hyrax_record.member_of_collections.each_with_index do |collection, i|
-          self.parsed_metadata["collection_#{i + 1}"] = collection.send(work_identifier)&.first
+          self.parsed_metadata["collection_#{i + 1}"] = collection.id
+          #     self.parsed_metadata["collection_#{i + 1}"] = collection.send(work_identifier)&.first
         end
-      end        
+      end
```
Comment (alishaevn) [inline app/models/bulkrax/csv_entry.rb:92]: @sephirothkod and @orangewolf bringing your attention here. the code we merged into `main` earlier doesn't work. I'm using the singular "collection" version of the code that keeps with what braydon was working on, but I'm not using the "send" method because it gave me the error below.
this works for BL and since braydon is deploying OD from his branch, figured this is ok to go into `main` for the time being.
Screen Shot 2021-09-09 at 11 05 36 PM

**Review by orangewolf — APPROVED** [review]
